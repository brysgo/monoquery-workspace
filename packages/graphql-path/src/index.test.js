import gql from "./";
import { print } from "graphql/language";

describe("graphqlPath", () => {
  it("returns a map of paths in a graphql query", () => {
    const fooFragment = gql`
      fragment Foo on Bar {
        blah
      }
    `;
    const { parsedQuery, fragmentNames, fragmentPaths } = gql`
      query FooQuery {
        ${"somethingOnRoot"}
        someResource {
          ${"onSomeResource"}
          anotherField {
            ${"onAnotherField"}
            ...Foo
          }
        }
      }

      ${fooFragment}
    `;
    expect(print(parsedQuery)).toMatchSnapshot();
    expect(fragmentNames.get(fooFragment)).toEqual("Foo");
    expect(fragmentPaths).toEqual({
      Foo: "someResource.anotherField"
    });
  });

  it("also supports graphql tag result for fragments", () => {
    const { parsedQuery: fooFragment } = gql`
      fragment Foo on Bar {
        blah
      }
    `;
    const { parsedQuery, fragmentNames, fragmentPaths } = gql`
      query FooQuery {
        ${"somethingOnRoot"}
        someResource {
          ${"onSomeResource"}
          anotherField {
            ${"onAnotherField"}
            ...Foo
          }
        }
      }

      ${fooFragment}
    `;
    expect(print(parsedQuery)).toMatchSnapshot();
    expect(fragmentNames.get(fooFragment)).toEqual("Foo");
    expect(fragmentPaths).toEqual({
      Foo: "someResource.anotherField"
    });
  });

  it("doesn't try to get the path of invalid strings", () => {
    const unparsedFragment = `
      fragment UnparsedFragment on Query {
        hello
      }
    `;
    const invalidGraphql = `
      foo
      bar
      baz
    `;
    const { parsedQuery, fragmentNames, fragmentPaths } = gql`
      query FooQuery {
        ...UndefinedFragment
        invalidWithoutInterpolated {
          ${invalidGraphql}
        }
      }
      ${unparsedFragment}
    `;
    expect(print(parsedQuery)).toMatchSnapshot();
    expect(fragmentNames.get(unparsedFragment)).toEqual();
    expect(fragmentPaths).toEqual({
      UndefinedFragment: ""
    });
  });

  it("works with nested fragments", () => {
    const subChildFragment = gql`
      fragment SubChild on Child {
        blah
      }
    `;
    const childFragment = gql`
      fragment Child on Base {
        anotherField {
          ...SubChild
        }
      }
      ${subChildFragment}
    `;
    const baseFragment = gql`
      fragment Base on Query {
        someResource {
          ...Child
        }
      }
      ${childFragment}
    `;
    const { parsedQuery, fragmentNames, fragmentPaths } = gql`
      query FooQuery {
        ...Base
      }
      ${baseFragment}
    `;
    expect(print(parsedQuery)).toMatchSnapshot();
    expect(fragmentNames.get(baseFragment)).toEqual("Base");
    expect(fragmentNames.get(childFragment)).toEqual("Child");
    expect(fragmentNames.get(subChildFragment)).toEqual("SubChild");
    expect(fragmentPaths).toEqual({
      Base: "",
      Child: "someResource",
      SubChild: "someResource.anotherField"
    });
  });

  it("is itself a valid graphql document", () => {
    const subChildFragment = gql`
      fragment SubChild on Child {
        blah
      }
    `;
    const childFragment = gql`
      fragment Child on Base {
        anotherField {
          ...SubChild
        }
      }
      ${subChildFragment}
    `;
    const baseFragment = gql`
      fragment Base on Query {
        someResource {
          ...Child
        }
      }
      ${childFragment}
    `;
    const query = gql`
      query FooQuery {
        ...Base
      }
      ${baseFragment}
    `;

    expect(print(query)).toMatchSnapshot();
  });
});
