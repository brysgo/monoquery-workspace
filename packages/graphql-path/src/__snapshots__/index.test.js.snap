// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`graphqlPath also supports graphql tag result for fragments 1`] = `
"query FooQuery {
  somethingOnRoot
  someResource {
    onSomeResource
    anotherField {
      onAnotherField
      ...Foo
    }
  }
}

fragment Foo on Bar {
  blah
}
"
`;

exports[`graphqlPath doesn't try to get the path of invalid strings 1`] = `
"query FooQuery {
  ...UndefinedFragment
  invalidWithoutInterpolated {
    foo
    bar
    baz
  }
}

fragment UnparsedFragment on Query {
  hello
}
"
`;

exports[`graphqlPath is itself a valid graphql document 1`] = `
"query FooQuery {
  ...Base
}

fragment Base on Query {
  someResource {
    ...Child
  }
}

fragment Child on Base {
  anotherField {
    ...SubChild
  }
}

fragment SubChild on Child {
  blah
}
"
`;

exports[`graphqlPath returns a map of paths in a graphql query 1`] = `
"query FooQuery {
  somethingOnRoot
  someResource {
    onSomeResource
    anotherField {
      onAnotherField
      ...Foo
    }
  }
}

fragment Foo on Bar {
  blah
}
"
`;

exports[`graphqlPath works with nested fragments 1`] = `
"query FooQuery {
  ...Base
}

fragment Base on Query {
  someResource {
    ...Child
  }
}

fragment Child on Base {
  anotherField {
    ...SubChild
  }
}

fragment SubChild on Child {
  blah
}
"
`;
